File: app/src/main/java/com/diarmaidlindsay/koohii/model/Meaning.java
package com.diarmaidlindsay.koohii.model;

/**
 * Represent an entry in the meaning table
 */
public class Meaning {
    private int id;
    private int heisigId;
    private String meaningText;

    public Meaning(int id, int heisigId, String meaningText) {
        this.id = id;
        this.heisigId = heisigId;
        this.meaningText = meaningText;
    }

    public int getId() {
        return id;
    }

    public int getHeisigId() {
        return heisigId;
    }

    public String getMeaningText() {
        return meaningText;
    }
}

File: app/src/main/java/tech/diarmaid/koohiiaite/database/entity/Meaning.kt
package tech.diarmaid.koohiiaite.database.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

/**
 * Represent an entry in the meaning table
 */
@Entity(tableName = "meaning",
        foreignKeys = [ForeignKey(entity = HeisigKanji::class, parentColumns = ["id"], childColumns = ["heisig_id"])])
data class Meaning(@PrimaryKey val id: Int,
                   @ColumnInfo(name = "heisig_id") val heisigId: Int,
                   @ColumnInfo(name = "meaning_text") val meaningText: String)
